test/LiveServer.hs

```diff
@@ -753,3 +754,38 @@ presignedPostPolicyFunTest = funTestWithBucket "Presigned Post Policy tests" $
         req' <- Form.formDataBody parts' req
         mgr <- NC.newManager NC.tlsManagerSettings
         NC.httpLbs req' mgr
+
+bucketPolicyFunTest :: TestTree
+bucketPolicyFunTest = funTestWithBucket "Bucket Policy tests" $
+  \step bucket -> do
+
+    step "bucketPolicy basic test - no policy exception"
+    resE <- MC.try $ getBucketPolicy bucket
+    case resE of
+      Left exn -> liftIO $ exn @?= ServiceErr "NoSuchBucketPolicy" "The bucket policy does not exist"
+      _        -> return ()
+
+    resE <- MC.try $ setBucketPolicy bucket T.empty
```

Comment:
Please don't reuse variable names - rename this to `resE1` and the next one below to `resE2`. This will avoid adding new shadow variable warnings. The existing ones are fixed via https://github.com/minio/minio-hs/pull/83